cpp/include/cudf/ipc.hpp

```diff
@@ -9,6 +10,8 @@ class CudaMessageReader : arrow::ipc::MessageReader {
   static std::unique_ptr<arrow::ipc::MessageReader> Open(arrow::cuda::CudaBufferReader* stream,
                                                          arrow::io::BufferReader* schema);
 
+  arrow::Result<std::unique_ptr<arrow::ipc::Message>> ReadNextMessage() override;
+
   arrow::Status ReadNextMessage(std::unique_ptr<arrow::ipc::Message>* message);
```

Comment:
Not specifically related to this PR, but this function signature with a raw pointer to a `unique_ptr` is a bit strange (mutable reference would be more idiomatic). From the context here, it seems this is because `arrow::ipc::ReadMessage` takes the message parameter as such, but browsing through the Arrow repo, I wasn't able to find this exact definition of `ReadMessage`. Any idea what's going on here?